code/modules/mob/living/simple_animal/vore/rous.dm

```diff
@@ -41,6 +41,121 @@
 	vore_pounce_chance = 45
 	vore_icons = SA_ICON_LIVING | SA_ICON_REST
 
-/mob/living/simple_animal/hostile/rous/death()
+/mob/living/simple_animal/hostile/bigrat/death()
 	playsound(src, 'sound/effects/mouse_squeak_loud.ogg', 50, 1)
+	if(resting)
+		resting = 0
+		icon_state = icon_dead
+		update_icon()
 	..()
+
+/mob/living/simple_animal/hostile/bigrat/passive
+	name = "curious giant rat"
+	desc = "In what passes for a hierarchy among verminous rodents, this one is king. It seems to be more interested on scavenging."
+	follow_dist = 1
+	var/mob/living/carbon/human/food
+	var/hunger = 0
+
+/mob/living/simple_animal/hostile/bigrat/passive/Life()
+	. = ..()
+	if(!. || ai_inactive) return
+
+	for(var/obj/item/weapon/reagent_containers/food/snacks/S in oview(src,7)) //Accept thrown offerings and scavenge surroundings.
+		if(isturf(S.loc))
+			if(get_dist(src,S) <=1)
+				visible_emote("hungrily devours \the [S].")
+				playsound(src.loc,'sound/items/eatfood.ogg', rand(10,50), 1)
+				qdel(S)
+				hunger = 0
+				food = null
+			else
+				WanderTowards(S.loc)
+			break
+
+	if(!food) return
+
+	var/food_dist = get_dist(src,food)
+
+	if(food_dist > 7) //Lose interest on this person.
+		food = null
+		if(hunger < 25) //Get miffed but don't let people evade stack you into pouncing the next victim on sight.
+			hunger += 5
+		else if(hunger > 25) //Leave room for interaction with the next victim if blueballed by evader.
+			hunger = 25
```

Comment:
Can replace all of this with `hunger = Clamp(hunger+5, 0, 25)` which sets hunger equal to hunger+5, with a min possible of 0, and a max possible of 25.